### PR TITLE
[buddy-lsp-server] add buddy-lsp-server

### PR DESCRIPTION
**Pull request by Origami404**
Having an LSP server is very handy when writing MLIR files. This PR builds an lsp server that supports the new dialect by adding Buddy-MLIR's new dialect directly to the registry.

In theory this program should be a drop-in replacement for mlir-lsp-server. Meaning you should be able to directly modify VSCode settings (or similar settings for other editors) to:

```json
{
    "mlir.server_path": "YOUR_BUDDY_MLIR_BUILD/bin/buddy-lsp-server",
}
```

After modification, your editor should have correct completion and error prompts for new dialects such as `rvv`.

### CODE REVIEW

**Comment by Origami404** [comment]
After #123 be merged, support for Gemini dialect needs to be added into this too.

File: tools/buddy-lsp-server/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+add_llvm_tool(buddy-lsp-server
+  buddy-lsp-server.cpp
+)
+
+target_link_libraries(buddy-lsp-server
+  PRIVATE
+  ${dialect_libs}
+  ${conversion_libs}
+  MLIRLspServerLib
+)
```
Comment (zhanghb97) [inline tools/buddy-lsp-server/CMakeLists.txt:13]: Please add an empty line here.

File: tools/buddy-lsp-server/buddy-lsp-server.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+void registerTestDialect(DialectRegistry &);
+void registerTestTransformDialectExtension(DialectRegistry &);
+} // namespace test
+#endif
+
+int main(int argc, char **argv) {
+  DialectRegistry registry;
+  registerAllDialects(registry);
+  // clang-format off
+  registry.insert<buddy::bud::BudDialect,
```
Comment (zhanghb97) [inline tools/buddy-lsp-server/buddy-lsp-server.cpp:49]: Please add the Gemmini dialect here.

**Review by zhanghb97 — CHANGES_REQUESTED** [review]
Cool! How about adding the how-to guide to the README?

**Comment by Origami404** [comment]
Rebased onto main and added instructions to README.

**Pull request merged**